### PR TITLE
Complex FFT support for KISS and  Accelerate

### PR DESCRIPTION
**Pull request by pabloEntropia**
This PR adds support to perform complex FFT and IFFT under Kiss or Accelerate.

Additionally, the issue about the IFFT normalization addressed in #772 is implemented here.

The [Panning](http://essentia.upf.edu/documentation/reference/std_Panning.html) algorithm is also modified to fit the new behavior of IFFT. 

All the relevant unit test have been updated and run and testified both under FFTW and KISS. However the tests **must** also be passed under Accelerate before merging.

### CODE REVIEW

**Comment by carthach** [comment]
Is this ever going to be merged in?

**Comment by dbogdanov** [comment]
This PR should be tested for the Accelerate case first. If you can verify it all works we can merge it anytime soon.

**Comment by pabloEntropia** [comment]
Also I we should remove `ConstantQ` and  `Chromagram` from the list of ignored algorithms once Accelarate it is checked, https://github.com/MTG/essentia/blob/088c9fa2649b2ca1a5d5ba05eda2ec69316d6687/src/wscript#L195

File: src/algorithms/standard/fftacomplex.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  void compute();
+  void configure();
+
+  static const char* name;
+  static const char* category;
+  static const char* description;
+
+ protected:
+  friend class IFFTAComplex;
+  static ForcedMutex globalFFTAcomplexMutex;
```
Comment (carthach) [inline src/algorithms/standard/fftacomplex.h]: This should be globalFFTAMutex

File: src/algorithms/standard/ifftacomplex.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#include "ifftacomplex.h"
+#include "ifftacomplex.h"
```
Comment (carthach) [inline src/algorithms/standard/ifftacomplex.cpp]: This should be just #include "fftacomplex.h" (no 'i')

File: src/algorithms/standard/ifftacomplex.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  }
+}
+
+void IFFTAComplex::configure() {
+  createFFTObject(parameter("size").toInt());
+  _normalize = parameter("normalize").toBool();
+}
+
+void IFFTAComplex::createFFTObject(int size) {
+  ForcedMutexLocker lock(FFTA::globalFFTAMutex);
```
Comment (carthach) [inline src/algorithms/standard/ifftacomplex.cpp]: ForcedMutexLocker lock(FFTAComplex::globalFFTAMutex);

**Comment by carthach** [comment]
I've compiled with Accelerate and passed the basetests and fft fftc units. @pabloEntropia there were a few inconsistencies between the KISS and Accelerate above which I needed to fix to compile. Maybe you could fix them and merge from master when you get a chance then I can re-run?

**Comment by pabloEntropia** [comment]
Thank you for your feedback @carthach . I have included your suggested modifications. Could you also report the results of the `ifft` and `ifftc` units with Accelerate?

**Pull request merged**